api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java

```diff
@@ -58,20 +59,26 @@
 	void deleteDiagnosis(Diagnosis diagnosis) throws DAOException;
 	
 	/**
-	 * Gets all diagnoses for a given encounter
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
+	 * primary or confirmed diagnoses.
 	 *
-	 * @param encounter the specific encounter to get the diagnoses for.
-	 * @return list of diagnoses for an encounter
+	 * @param encounter the encounter for which to fetch diagnoses
+	 * @param primaryOnly whether to return only primary diagnoses
+	 * @param confirmedOnly whether to return only confirmed diagnoses
+	 * @return the list of (primary, confirmed) diagnoses for the given encounter
 	 */
-	List<Diagnosis> getDiagnoses(Encounter encounter);
```

Comment:
thanks @hgrsd   are you considering things like backward compatibility? Don't you think that removing this will break the modules that depend on it?

Reply:
Hi @HerbertYiga! Thanks for your comment. From what I understood from the developer guidelines, DAO methods can be removed without marking them as deprecated because no external modules must depend on them. The method wasn't used by any code in `openmrs-core`. Is that right?

Reply:
They shouldn't depend on DAO classes but we have zero guarantees that they aren't :)

Reply:
Thanks. Clearly I was too enthusiastic getting rid of it - I'll bring it back for backwards compat.

Reply:
FYI, here is the docs I used which pointed to DAO methods being OK to remove: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation. I wonder if it's worth updating it to discourage doing so?